src/main/java/edu/umich/ctools/sectionsUtilityTool/SISSupportProcess.java

```diff
@@ -0,0 +1,211 @@
+
+		int status = arw.getStatus();
+		if (status == Utils.API_EXCEPTION_ERROR || status == Utils.API_UNKNOWN_ERROR) {
+			csvMap.put(status, arw.getMessage());
+			return csvMap;
+		}
+		if (status != HttpStatus.SC_OK) {
+			csvMap.put(status, arw.getApiResp());
+			return csvMap;
+		}
+		JSONObject courseJson = new JSONObject(arw.getApiResp());
+		String courseSisId = (String) courseJson.get("sis_course_id");
+
+		String[] sections = csvSectionsData.split("\n");
+		StringBuilder completeCsvFile = new StringBuilder();
+		completeCsvFile.append(Utils.CSV_HEADERS_SECTIONS);
+		completeCsvFile.append("\n");
+		String sisSectionIdChunk = Utils.getSisSectionIDChunk(request);
+		for (int i = 1; i < sections.length; i++) {
+			String[] aSectionInfo = sections[i].split(",");
+			String section = aSectionInfo[0].replace(aSectionInfo[0], sisSectionIdChunk + aSectionInfo[0]) + ","
+					+ aSectionInfo[1] + "," + "active" + "," + courseSisId;
+			completeCsvFile.append(section);
+			completeCsvFile.append("\n");
+		}
+		M_log.debug(completeCsvFile);
+		csvMap.put(status, completeCsvFile.toString());
+		return csvMap;
+	}
+
+	private ApiResultWrapper sisUploadCall(String csvFileContent){
+		String url = Utils.urlConstructor("/accounts/1/sis_imports?batch_mode=0&override_sis_stickiness=0&extension=csv");
+		HttpPost post = new HttpPost(url);
+		post.setEntity(new StringEntity(csvFileContent, ContentType.create(Utils.CONSTANT_MIME_TEXT_CSV, "UTF-8")));
+		ApiResultWrapper arw = Utils.makeApiCall(post);
+		M_log.debug("API response: "+arw.getApiResp());
+		return arw;
+	}
+
+	private ApiResultWrapper getCourseSisId(String courseId) {
+		M_log.debug("getCourseSisId() call");
+		String course_account_id;
+		String url = Utils.urlConstructor("/courses/", courseId);
+		ApiResultWrapper arw = Utils.makeApiCall(new HttpGet(url));
+		String msg = String.format("couldn't get course_sis_id for course %s, with Status code %s, due to: %s", courseId,
+				arw.getStatus(), (arw.getApiResp().isEmpty()) ? arw.getMessage() : arw.getApiResp());
+
+		if(!Utils.didApiReturnedWithOutErrors(msg, arw)){
+			return arw;
+		}
+		JSONObject courseJson = new JSONObject(arw.getApiResp());
+		if (courseJson.isNull("sis_course_id")) {
+			return createACourseSISId(courseId);
+		}
+		return arw;
+	}
+
+	private ApiResultWrapper createACourseSISId(String courseId) {
+		M_log.debug("createACourseSISId() call");
+		String courseSISId = Utils.generateSisCourseId();
+		//https://umich.test.instructure.com/api/v1/courses/183787/?course[sis_course_id]=pu_sis_id_0329_3
+		String url = Utils.urlConstructor("/courses/", courseId, "/?course[sis_course_id]=" + courseSISId);
+		ApiResultWrapper arw = Utils.makeApiCall(new HttpPut(url));
+		String msg = String.format("couldn't create course_sis_id for course %s, with status code %s, due to: %s", courseId,
+				arw.getStatus(), (arw.getApiResp().isEmpty()) ? arw.getMessage() : arw.getApiResp());
+
+		 if(!Utils.didApiReturnedWithOutErrors(msg, arw)){
```

Comment:
Might as well just return arw without the check since it will be returned in any case.